### DISCUSSION TITLE
Cylinder parameters

### DISCUSSION THREAD

**Discussion by Kbourbat25** [PMEAL/porespy#617] [Q&A] [answered]
Hello,
I want to use porespy to generate a fibrous medium with fixed porosity and enventually generate .m file compatible with Comsol. 
The definition of cylinder under comsol sowtware requires to know the radius, center of the base of cylinder, phi and theta as definied in porespy. The difficulty for me is how to obtain all this informations and what is the variables to call and how to call them?

Thank you for your help.

**Comment by jgostick** [comment]
Hi @Kbourbat25, our method just returns a voxel image of cylinders, BUT we do this by first generating lines then fattening them.  So maybe you could repurpose our 'line generator' code?  There are two functions in the generators module, the public ``cylinders`` function which internally calls a private ``_cylinders`` function in order to match the given porosity.  You can see how we add fibers as single lines by looking at ``_cylinders`` here: https://github.com/PMEAL/porespy/blob/4a786d753840a5ea2d41cdacf83f0c5010120732/porespy/generators/_imgen.py#L832

Meanwhile, it might be worth considering adding a comsol capatible cylinders generator.  Could you possible post here the file format that comsol wants?

**Reply by Kbourbat25** [reply, marked as answer]
Hi @jgostick , thank you for this reply, you found in what follow the difinition of cylinder under comsol :
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
% The difinition of cylinder in Comsol
% name cyl1 , type Cylinder
model.component('comp1').geom('geom1').create('cyl1', 'Cylinder');
% radius r=0.01
model.component('comp1').geom('geom1').feature('cyl1').set('r', 0.01);
% position of the center of the base of cylinder 
model.component('comp1').geom('geom1').feature('cyl1').set('pos', [0 0 0]);
% height of the cylinder h=2 
model.component('comp1').geom('geom1').feature('cyl1').set('h', 2);
% coordinate system = spherical
model.component('comp1').geom('geom1').feature('cyl1').set('axistype', 'spherical');
% angle theta=45 angle between axis of cylinder and z axis.
% angle phi=30 angle between axis of cylinder and x axis.
model.component('comp1').geom('geom1').feature('cyl1').set('ax2', [45 30]);
% run 
model.component('comp1').geom('geom1').run('cyl1');
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
Best regards
